walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/engine/serailising/JsonRpcSerializer.kt

```diff
@@ -0,0 +1,17 @@
+package org.walletconnect.walletconnectv2.engine.serailising
+
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.Moshi
+
+class JsonRpcSerializer(val moshi: Moshi) : JsonRpcSerialising {
+
+    override fun <T> trySerialize(typeClass: Class<T>, type: T): String =
+        moshi.adapter(typeClass).toJson(type)
+
+    override fun <T> tryDeserialize(type: Class<T>, json: String): T? =
+        try {
+            org.walletconnect.walletconnectv2.moshi.adapter(type).fromJson(json)
+        } catch (error: JsonDataException) {
+            null
+        }
```

Comment:
Remove the interface and replace the `tryDeserialize` function with an inline function

```suggestion
        inline fun <reified T> tryDeserialize(json: String): T? {
        return runCatching { moshi.adapter(T::class.java).fromJson(json) }.getOrNull()
    }
```

Reply:
Good call. I like this approach with inline function and refined keyword more.